woocommerce/woocommerce
# Fix Analytics Dashboard rendering for malformed preferences

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





- [Add a more descriptive error](https://github.com/woocommerce/woocommerce/commit/9f7dd033487597033a6426be7554ab5213bf203d) 
   when `woocommerce_dashboard_default_sections` returns malformed results.
- [Use defaults when `userPreferences.dashboard_sections` is not an array](https://github.com/woocommerce/woocommerce/commit/d473ded3ddef75f783b5315e5f2f5c04df2b7fad) 

Fix https://github.com/woocommerce/woocommerce/issues/51844



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add a `woocommerce_dashboard_default_sections` filter that corrupts the default setting
	```php
	add_action( 'admin_enqueue_scripts', function () {
		wp_add_inline_script( 'wp-hooks', <<<EOD
	    wp.hooks.addFilter( 'woocommerce_dashboard_default_sections', 'dev-blog-example', function( sections ) {
	        console.log(sections);
	        return {};
	    } );
	    EOD );
	} );
	```
2. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Foverview` check that the error message is more descriptive than on `trunk`
3. Disable the above filter
4. Enable new product editor
5. And edit any product
6. Open kebab menu and pick "Show developer tools"
   ![Image](https://github.com/user-attachments/assets/8aa6fadf-1fa3-4662-ab2d-bb5e50ed5ef7)
7. Open regular browser dev tools, execute
	```js
	window.__wcbt.updateUserPreferences( {...window.__wcbt.userPreferences,dashboard_sections:{}} )
	```
8. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Foverview`
9. Check that the dashboard renders correctly using the default settings




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
